### PR TITLE
new test for network sriov device test

### PR DESCRIPTION
**Pull request by smruti77**
test to add and remove sriov logical device.

Signed-off-by: bismurti bidhibrata pattajoshi <bbidhibr@in.ibm.com>

### CODE REVIEW

**Comment by smruti77** [comment]
[sriov_device.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4578962/sriov_device.log)

File: io/net/sriov_device_test.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+            self.run_command(cmd)
+        except CommandFailed as cf:
+            self.fail("failed to create logical device" % cf)
+        cmd = "lshwres -r sriov --rsubtype logport -m  %s \
+               --level eth | grep %s | grep %s" \
+               % (self.server, self.lpar, self.mac_id)
+        output = self.run_command(cmd)
+        self.logical_port_id = output.split(',')[6].split('=')[-1]
+
+    def removed_logical_device(self):
```
Comment (abdhaleegit) [inline io/net/sriov_device_test.py:189]: change `removed_logical_device` to `test_remove_sriov_device`
make it a test to help in scenario test

File: io/net/sriov_device_test.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        con.expect(con.PROMPT, timeout=timeout)
+        output = "".join(con.before)
+        con.sendline("echo $?")
+        con.prompt(timeout)
+        exitcode = int(''.join(con.before.splitlines()[1:]))
+        if exitcode != 0:
+            raise CommandFailed(command, output, exitcode)
+        return output
+
+    def add_logical_sriov(self):
```
Comment (abdhaleegit) [inline io/net/sriov_device_test.py:170]: make it a test, change `add_logical_sriov` to `tests_add_sriov_device`

File: io/net/sriov_device_test.py [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+    def removed_logical_device(self):
+        """
+        remove logical device
+        """
+        cmd = 'chhwres -r sriov -m %s --rsubtype logport \
+              -o r -p %s -a \"adapter_id=%s,logical_port_id=%s\" ' \
+              % (self.server, self.lpar, self.adapter_id,
+                 self.logical_port_id)
+        try:
```
Comment (abdhaleegit) [inline io/net/sriov_device_test.py:197]: check if the logical device is really removed, using lshwres command

File: io/net/sriov_device_test.py [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+    '''
+    adding and deleting logical sriov device through
+    HMC.
+    '''
+    def setUp(self):
+        '''
+        set up required packages and gather necessary test inputs
+        '''
+        smm = SoftwareManager()
+        packages = ['rsct.basic', 'rsct.core.utils',
```
Comment (abdhaleegit) [inline io/net/sriov_device_test.py:63]: src, and other packages

File: io/net/sriov_device_test.py [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+        cmd = 'lssyscfg -r sys  -F name'
+        output = self.run_command(cmd)
+        self.server = ''
+        for line in output.splitlines():
+            if line in self.lpar:
+                self.server = line
+                break
+        if not self.server:
+            self.cancel("Managed System not got")
+        self.sriov_adapter = self.params.get('sriov_adapter', '*', default=None)
```
Comment (abdhaleegit) [inline io/net/sriov_device_test.py:86]: can we have multiple adapters to create more than one sriov device ?

Comment (abdhaleegit) [reply io/net/sriov_device_test.py:86]: make sure multiple sriov_adapter input is space separated

we need multiple sriov in cases where bonding is tested, so sriov device coming from 2 different adapter port

File: io/net/sriov_device_test.py [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+        self.sriov_port = self.params.get('sriov_port', '*', default=None)
```
Comment (abdhaleegit) [inline io/net/sriov_device_test.py:87]: similarlry multiple space separated port

File: io/net/sriov_device_test.py [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+        self.ipaddr = self.params.get('ipaddr', '*', default="")
+        self.netmask = self.params.get('netmask', '*', default="")
+        self.peer_ip = self.params.get('peer_ip', '*', default="")
+        self.mac_id = self.params.get('mac_id',
+                                      default="02:03:03:03:03:01")
```
Comment (abdhaleegit) [inline io/net/sriov_device_test.py:92]: similarlry for all above inputs, multiple space separated inputs

File: io/net/sriov_device_test.py (hunk 1 above)
Comment (smruti77) [reply io/net/sriov_device_test.py:189]: ok, then in test add  i will add ping ..?

**Comment by smruti77** [comment]
[sriov_test.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4586416/sriov_test.log)

File: io/net/sriov_device_test.py [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+            if not self.list_device(mac):
+                self.fail("failed to list logical device after add operation")
+            device = self.find_device(mac)
+            networkinterface = NetworkInterface(device, self.local)
+            networkinterface.add_ipaddr(ipaddr, netmask)
+            networkinterface.bring_up()
+            if networkinterface.ping_check(peer_ip, count=5) is not None:
+                self.fail("ping check failed")
+
+    def test_removed_logical_device(self):
```
Comment (abdhaleegit) [inline io/net/sriov_device_test.py:189]: removed = remove

**Comment by smruti77** [comment]
[multiple_logical_device.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4590657/multiple_logical_device.log)
[single_logical_device.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4590658/single_logical_device.log)

**Comment by abdhaleegit** [comment]
LGTM, please merge

**Pull request merged**